### ISSUE TITLE
Podman run blocks in `isPathOnVolume` function forever

### ISSUE THREAD

**Issue by bacher09** [containers/podman#10216]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

There is a bug in current [isPathOnVolume](https://github.com/containers/podman/blob/master/libpod/container_path_resolution.go#L125) function which in some cases leads to infinite loop which causes some commands to stuck. This happens cause `filepath.Dir(dest)` could return `.` in some cases and calling same method again would return same result (`.`).

**Steps to reproduce the issue:**

1. Create a Docker image using this `Dockerfile`:
```
cat Dockerfile 
FROM alpine:3.12
VOLUME ['/etc/foo']
WORKDIR /etc/foo
CMD ["echo", "test"]
```
it can be compiled using any tool (I used original `docker build`).
**NOTE**: that `VOLUME` uses single quotes `'` and not `"`.

2. Pull that image into your system and make it available for `podman`

3. Try to start that container using that image:
```
podman run testimg
```
and this would stuck forever in the loop.

**Describe the results you received:**

`podman run testimg` got stuck consuming cpu resources.

**Describe the results you expected:**

It should either fail or succeed. Interestingly spec like this works fine in a `docker`, it also used to work in a `podman` version 2.

**Extra information**

Basically using single quotes causes `vol.Dest` to become `[/etc/foo]` string and calling `filepath.Dir` on it will eventually return `[` and then `.` so will never match `/`.

I skipped some fields cause these are not really necessary cause issue is really issue to reproduce.

**Comment by bacher09** [comment]
The simplest fix would be something like this:
```
diff --git a/libpod/container_path_resolution.go b/libpod/container_path_resolution.go
index d798963b1..ec7306ca1 100644
--- a/libpod/container_path_resolution.go
+++ b/libpod/container_path_resolution.go
@@ -128,7 +128,7 @@ func isPathOnVolume(c *Container, containerPath string) bool {
                if cleanedContainerPath == filepath.Clean(vol.Dest) {
                        return true
                }
-               for dest := vol.Dest; dest != "/"; dest = filepath.Dir(dest) {
+               for dest := vol.Dest; dest != "/" && dest != "."; dest = filepath.Dir(dest) {
                        if cleanedContainerPath == dest {
                                return true
                        }
```
which should ensure that it would never loop forever in that code. But also IMHO it would be nice to have some validation for `volume` as well.

**Comment by mheon** [comment]
@vrothberg PTAL

**Comment by rhatdan** [comment]
@bacher09 Please open a PR to fix this?
